File: scenarios/long_chain_dag/fixtures/write_marker.py
from __future__ import annotations

import json
import sys
from pathlib import Path


def main() -> None:
    workspace = Path(sys.argv[1])
    marker = sys.argv[2]
    workspace.mkdir(parents=True, exist_ok=True)
    (workspace / f"{marker}.txt").write_text(marker, encoding="utf-8")
    print(json.dumps({"marker": marker}))


if __name__ == "__main__":
    main()

File: scenarios/long_chain_dag/fixtures/write_marker.ts
import { mkdir, writeFile } from "node:fs/promises";
import path from "node:path";

const workspace = process.argv[2];
const marker = process.argv[3];
if (!workspace || !marker) {
  throw new Error("workspace and marker are required");
}

await mkdir(workspace, { recursive: true });
await writeFile(path.join(workspace, `${marker}.txt`), marker, "utf8");
console.log(JSON.stringify({ marker }));
